### DISCUSSION TITLE
Confirm Actions?

### DISCUSSION THREAD

**Discussion by tonypartridge** [rappasoft/laravel-livewire-tables#952] [Q&A] [answered]
We have a few options which we really want to have people confirm that they really want to perform this action... The problem is I cannot see a native way to do this as we have no control over the view. For example, I would usually include a modal, then in the component a isConfirmed function... I was able to call the parent function to trigger the modal view when including it in the parent component. But then struggle to pass is down to the table component :/

**Comment by tonypartridge** [comment, marked as answer]
I've resolved this with the standard comfirm on the component functions and included the modal into a configurableArea in a custom view.

**Reply by rabol** [reply]
would you like to share, please ?

**Reply by tonypartridger** [reply]
Sure, add to the component:

public $showModal = false, $record;

Add wire:click="showModal({{ $record->id }}" method, then add a function: 

public showModal($id) {
    $this->showModal = true;
    $this->record = RecordsModel::findOrFail($id);
}

Add a function function on the modal to call wire:click="saveModalInfo()"

public function saveModalInfo() {
     Run all your usual save functions here

    if ($save) {
        $this->reset();
        $this->showModal = false;
    }
}


And ensure in the component config you have: 

@if($showModal) 
    Modal code here
@endif
